src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -142,13 +176,25 @@ const DateTimeBlock : React.FC<IProps> = ({
   useEffect(()=>{
     if(allowAfterMidnight && isEqual(date, endOfDay(date))){
       setDisplayHours('24');
-      setDisplayMinutes('00');
-    } else {
-      setDisplayMinutes(format(date, 'mm'));
-      setDisplayHours(format(date,'HH'));
+      setDisplayMinutes('0');
     }
   },[date, allowAfterMidnight]);
 
+  const onBlurInputs = useCallback(() =>{
+    if(displayHours === '0'){
+      setDisplayHours('00');
+    }
+    if(displayMinutes ==='0'){
+      setDisplayMinutes('00');
+    } 
+    if(displayHours === ''){
+      setDisplayHours('00');
+    }
+    if(displayMinutes ===''){
+      setDisplayMinutes('00');
+    }
+  }, [displayHours, displayMinutes]);
```

Comment:
You can use `||` operator here to combine these multiple if's.

Reply:
Done